smt_proofs_test.go

```diff
@@ -234,20 +253,52 @@ func TestSMT_ProveClosestEmptyAndOneNode(t *testing.T) {
 	path := sha256.Sum256([]byte("testKey2"))
 	flipPathBit(path[:], 3)
 	flipPathBit(path[:], 6)
-	closestPath, closestValueHash, proof, err = smt.ProveClosest(path[:])
+	proof, err = smt.ProveClosest(path[:])
+	require.NoError(t, err)
+	require.Equal(t, proof, &SparseMerkleClosestProof{
+		Path:         path[:],
+		FlippedBits:  []int{0},
+		Depth:        0,
+		ClosestPath:  placeholder(smt.Spec()),
+		ClosestProof: &SparseMerkleProof{},
+	})
+
+	result, err := VerifyClosestProof(proof, smt.Root(), NoPrehashSpec(sha256.New(), false))
 	require.NoError(t, err)
-	require.Equal(t, proof, &SparseMerkleProof{})
-
-	result := VerifyProof(proof, smt.Root(), closestPath, closestValueHash, NoPrehashSpec(sha256.New(), false))
 	require.True(t, result)
 
-	require.NoError(t, smt.Update([]byte("foo"), []byte("bar")))
-	closestPath, closestValueHash, proof, err = smt.ProveClosest(path[:])
+	require.NoError(t, smn.Stop())
+}
+
+func TestSMT_ProveClosestOneNode(t *testing.T) {
+	var smn KVStore
+	var smt *SMT
+	var proof *SparseMerkleClosestProof
+	var err error
+
+	smn, err = NewKVStore("")
 	require.NoError(t, err)
-	require.Equal(t, proof, &SparseMerkleProof{})
-	require.Equal(t, closestValueHash, []byte("bar"))
+	smt = NewSparseMerkleTree(smn, sha256.New(), WithValueHasher(nil))
 
-	result = VerifyProof(proof, smt.Root(), closestPath, closestValueHash, NoPrehashSpec(sha256.New(), false))
+	path := sha256.Sum256([]byte("testKey2"))
+	flipPathBit(path[:], 3)
+	flipPathBit(path[:], 6)
+
+	require.NoError(t, smt.Update([]byte("foo"), []byte("bar")))
```

Comment:
I'd move this to be below line 282 so it's alongside the code that prepare the tree.